horovod/mxnet/__init__.py

```diff
@@ -102,11 +102,10 @@ def __init__(self, params, optimizer, optimizer_params=None):
     def _allreduce_grads(self):
         if size() == 1: return
 
-        # sort needed for Python < 3.6 is not guaranteed
-        for i, param in enumerate(sorted(self._params, key=lambda p: p.name)):
+        for i, param in enumerate(self._params):
             if param.grad_req != 'null':
                 allreduce_(param.list_grad()[0], average=False,
-                           name=str(i), priority=-i)
+                           name=param.name, priority=-i)
```

Comment:
Do we need a sanity check of no same name here? Otherwise, we may end up with two allreduce operation with the same name?

Reply:
The backend will already generate an error for duplicate names if it detects them so we should not need an additional check here.